### DISCUSSION TITLE
JSON API - Authentication

### DISCUSSION THREAD

**Discussion by nheyser** [grafana/grafana-infinity-datasource#354] [Q&A]
Hey there,

I would like to pull data from the Tronity API.

The authentication flow is described [here](https://app.tronity.tech/docs#section/Authentication-Flow).

I need to get a Bearer Token with the following credentials
- Client ID
- Client Secret
- Token URL: https://api.tronity.tech/authentication

Where do I need to put the "grant_type"?
How to check whether the authentication works?

Nik

**Comment by yosiasz** [comment]
Maybe add it as endpoint parameter or header?

![image](https://user-images.githubusercontent.com/900951/179116035-d34eb16b-e152-444a-99eb-897613425356.png)

**Reply by nheyser** [reply]
It is not a full OAuth2 API, it uses only a token.

If I add the token as bearer auth, I can access the API, but how do I frequently refresh the bearer-token?

**Reply by yosiasz** [reply]
add it as a variable, the call to the token fetch.
